src/test/java/io/cdap/plugin/facebook/source/batch/FacebookRecordReaderTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.facebook.source.batch;
+
+import com.facebook.ads.sdk.APIException;
+import com.facebook.ads.sdk.APINodeList;
+import com.facebook.ads.sdk.AdsInsights;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import io.cdap.plugin.facebook.source.common.requests.InsightsRequest;
+import io.cdap.plugin.facebook.source.common.requests.InsightsRequestFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Iterator;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(fullyQualifiedNames = "io.cdap.plugin.facebook.source.common.requests.InsightsRequestFactory")
+public class FacebookRecordReaderTest {
+  private static final Gson gson = new GsonBuilder().create();
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testIterate() throws IOException, APIException {
```

Comment:
this test is not very readable and maintainable. What about replacing it with functional test?